django_project/trail/models/trail.py

```diff
@@ -0,0 +1,92 @@
+# coding=utf-8
+"""Model definitions for a trail app"""
+
+import os
+import uuid as uuid_lib
+
+from django.contrib.gis.db import models
+from django.utils.translation import ugettext_lazy as _
+from django.utils.text import slugify
+from django.conf.global_settings import MEDIA_ROOT
+
+from unidecode import unidecode
+
+from core.settings.contrib import STOP_WORDS
+# from ..utils import GUIDModel
+
+__author__ = 'Alison Mukoma <alison@kartoza.com>'
+__date__ = '01/09/2018'
+__license__ = "GPL"
+__copyright__ = 'kartoza.com'
+
+
+class Trail(models.Model):
+    "Model definition of a Trail."
+
+    name = models.CharField(
+        _('Name of Trail'),
+        max_length = 255,
+        help_text = _('Enter name of the Trail.')
+    )
+
+    guid = models.UUIDField(
+        _('GUID'),
+        db_index = False,
+        default=uuid_lib.uuid4,
+        editable=False
+    )
+
+    notes = models.TextField(
+        _("Notes on named Trail"),
+        max_length = 300,
+        null = True,
+        blank = True,
+        help_text = _('Enter some notes regarding the above named trail')
+    )
+
+    offset = models.CharField(
+        _("Offset"),
+        max_length=255,
+        null = True,
+        blank = True,
+        help_text = _('Enter offset value i.e -2')
+    )
+
+    colour = models.CharField(
+        _("Colour"),
+        max_length = 255,
+        null=True,
+        blank = True,
+        help_text = _('Enter colour of the trail.')
+    )
+
+    image = models.ImageField(
+        _('Image file'),
+        null=True,
+        blank=True,
+        upload_to=os.path.join(MEDIA_ROOT, 'images/trail'),
+        help_text = _(
+            'An image of the trail. '
+            'Most browsers support dragging the image directly on to the '
+            '"Choose File" button above.')
+    )
+
+    geometry = models.PointField()
```

Comment:
See if you can specify SRID=4326 here an that we want XYZ geometry support (z being altitude)